{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": "true"
   },
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Imports\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Imports</a></div><div class=\"lev1\"><a href=\"#Load-inputs\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Load inputs</a></div><div class=\"lev2\"><a href=\"#Jodie's-SNP-tables\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>Jodie's SNP tables</a></div><div class=\"lev2\"><a href=\"#Jostins-IBD-graph\"><span class=\"toc-item-num\">2.2&nbsp;&nbsp;</span>Jostins IBD graph</a></div><div class=\"lev1\"><a href=\"#File-contents\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>File contents</a></div><div class=\"lev2\"><a href=\"#Merck\"><span class=\"toc-item-num\">3.1&nbsp;&nbsp;</span>Merck</a></div><div class=\"lev2\"><a href=\"#Candidates\"><span class=\"toc-item-num\">3.2&nbsp;&nbsp;</span>Candidates</a></div><div class=\"lev1\"><a href=\"#Functions\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Functions</a></div><div class=\"lev2\"><a href=\"#Table-manipulation\"><span class=\"toc-item-num\">4.1&nbsp;&nbsp;</span>Table manipulation</a></div><div class=\"lev2\"><a href=\"#Working-with-the-graph\"><span class=\"toc-item-num\">4.2&nbsp;&nbsp;</span>Working with the graph</a></div><div class=\"lev1\"><a href=\"#Explore\"><span class=\"toc-item-num\">5&nbsp;&nbsp;</span>Explore</a></div><div class=\"lev2\"><a href=\"#Prune-candidates-table-to-just-some-specific-info\"><span class=\"toc-item-num\">5.1&nbsp;&nbsp;</span>Prune candidates table to just some specific info</a></div><div class=\"lev2\"><a href=\"#List-of-top-candidate-genes\"><span class=\"toc-item-num\">5.2&nbsp;&nbsp;</span>List of top candidate genes</a></div><div class=\"lev2\"><a href=\"#Which-of-the-top_candidate-genes-can-we-find-in-the-IBD-graph?\"><span class=\"toc-item-num\">5.3&nbsp;&nbsp;</span>Which of the top_candidate genes can we find in the IBD graph?</a></div><div class=\"lev3\"><a href=\"#Can-we-find-paths-between-any-of-the-genes?\"><span class=\"toc-item-num\">5.3.1&nbsp;&nbsp;</span>Can we find paths between any of the genes?</a></div><div class=\"lev3\"><a href=\"#Lets-compare-the-symptoms-from-subjects-with-the-SNPs-in-the-genes-we-found-that-have-paths-linking-them?\"><span class=\"toc-item-num\">5.3.2&nbsp;&nbsp;</span>Lets compare the symptoms from subjects with the SNPs in the genes we found that have paths linking them?</a></div><div class=\"lev3\"><a href=\"#Any-familes-have-SNPs-in-any-of-the-other-genes-in-that-path?\"><span class=\"toc-item-num\">5.3.3&nbsp;&nbsp;</span>Any familes have SNPs in any of the other genes in that path?</a></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:59:35.144044",
     "start_time": "2016-04-27T11:59:35.134198"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import itertools\n",
    "\n",
    "import networkx as nx\n",
    "import networkxgmml\n",
    "\n",
    "import pandas as pd\n",
    "pd.options.display.max_columns = 999\n",
    "pd.options.display.max_colwidth = 1000\n",
    "\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:40.017445",
     "start_time": "2016-04-27T11:43:39.998291"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.11'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nx.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:40.178441",
     "start_time": "2016-04-27T11:43:40.020153"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ibd_ = \"/home/gus/MEGAsync/zim/main/BCH/Projects/Jodie_families/Jostins_et_al/jostins_2012_IBD_network.xgmml\"\n",
    "merk_ = \"/home/gus/MEGAsync/zim/main/BCH/Projects/Jodie_families/Excels_from_Jodie/WGS for 99 samples from Merck for Gus.xlsx\"\n",
    "cand_ = \"/home/gus/MEGAsync/zim/main/BCH/Projects/Jodie_families/Excels_from_Jodie/Snapper candidate tracking 1_12_2016 (3).xlsx\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Jodie's SNP tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:41.740484",
     "start_time": "2016-04-27T11:43:40.180611"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "merk = pd.read_excel(merk_)\n",
    "cand = pd.read_excel(cand_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Jostins IBD graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:41.995118",
     "start_time": "2016-04-27T11:43:41.742637"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ibd = networkxgmml.XGMMLReader(open(ibd_, 'rb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# File contents"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.145713",
     "start_time": "2016-04-27T11:43:41.997092"
    },
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inheritance_mode</th>\n",
       "      <th>project_id</th>\n",
       "      <th>family_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>rsid</th>\n",
       "      <th>annotation</th>\n",
       "      <th>exac_af</th>\n",
       "      <th>1kg_af</th>\n",
       "      <th>exac_popmax_af</th>\n",
       "      <th>merck_wgs_3793_af</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "      <th>genotype0_sample_id</th>\n",
       "      <th>genotype0_str</th>\n",
       "      <th>genotype0_num_alt</th>\n",
       "      <th>genotype0_allele_balance</th>\n",
       "      <th>genotype0_AD</th>\n",
       "      <th>genotype0_DP</th>\n",
       "      <th>genotype0_GQ</th>\n",
       "      <th>genotype0_PL</th>\n",
       "      <th>genotype1_sample_id</th>\n",
       "      <th>genotype1_str</th>\n",
       "      <th>genotype1_num_alt</th>\n",
       "      <th>genotype1_allele_balance</th>\n",
       "      <th>genotype1_AD</th>\n",
       "      <th>genotype1_DP</th>\n",
       "      <th>genotype1_GQ</th>\n",
       "      <th>genotype1_PL</th>\n",
       "      <th>genotype2_sample_id</th>\n",
       "      <th>genotype2_str</th>\n",
       "      <th>genotype2_num_alt</th>\n",
       "      <th>genotype2_allele_balance</th>\n",
       "      <th>genotype2_AD</th>\n",
       "      <th>genotype2_DP</th>\n",
       "      <th>genotype2_GQ</th>\n",
       "      <th>genotype2_PL</th>\n",
       "      <th>genotype3_sample_id</th>\n",
       "      <th>genotype3_str</th>\n",
       "      <th>genotype3_num_alt</th>\n",
       "      <th>genotype3_allele_balance</th>\n",
       "      <th>genotype3_AD</th>\n",
       "      <th>genotype3_DP</th>\n",
       "      <th>genotype3_GQ</th>\n",
       "      <th>genotype3_PL</th>\n",
       "      <th>genotype4_sample_id</th>\n",
       "      <th>genotype4_str</th>\n",
       "      <th>genotype4_num_alt</th>\n",
       "      <th>genotype4_allele_balance</th>\n",
       "      <th>genotype4_AD</th>\n",
       "      <th>genotype4_DP</th>\n",
       "      <th>genotype4_GQ</th>\n",
       "      <th>genotype4_PL</th>\n",
       "      <th>genotype5_sample_id</th>\n",
       "      <th>genotype5_str</th>\n",
       "      <th>genotype5_num_alt</th>\n",
       "      <th>genotype5_allele_balance</th>\n",
       "      <th>genotype5_AD</th>\n",
       "      <th>genotype5_DP</th>\n",
       "      <th>genotype5_GQ</th>\n",
       "      <th>genotype5_PL</th>\n",
       "      <th>genotype6_sample_id</th>\n",
       "      <th>genotype6_str</th>\n",
       "      <th>genotype6_num_alt</th>\n",
       "      <th>genotype6_allele_balance</th>\n",
       "      <th>genotype6_AD</th>\n",
       "      <th>genotype6_DP</th>\n",
       "      <th>genotype6_GQ</th>\n",
       "      <th>genotype6_PL</th>\n",
       "      <th>genotype7_sample_id</th>\n",
       "      <th>genotype7_str</th>\n",
       "      <th>genotype7_num_alt</th>\n",
       "      <th>genotype7_allele_balance</th>\n",
       "      <th>genotype7_AD</th>\n",
       "      <th>genotype7_DP</th>\n",
       "      <th>genotype7_GQ</th>\n",
       "      <th>genotype7_PL</th>\n",
       "      <th>genotype8_sample_id</th>\n",
       "      <th>genotype8_str</th>\n",
       "      <th>genotype8_num_alt</th>\n",
       "      <th>genotype8_allele_balance</th>\n",
       "      <th>genotype8_AD</th>\n",
       "      <th>genotype8_DP</th>\n",
       "      <th>genotype8_GQ</th>\n",
       "      <th>genotype8_PL</th>\n",
       "      <th>genotype9_sample_id</th>\n",
       "      <th>genotype9_str</th>\n",
       "      <th>genotype9_num_alt</th>\n",
       "      <th>genotype9_allele_balance</th>\n",
       "      <th>genotype9_AD</th>\n",
       "      <th>genotype9_DP</th>\n",
       "      <th>genotype9_GQ</th>\n",
       "      <th>genotype9_PL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A2</td>\n",
       "      <td>PTCHD1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>23397873</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs147324438</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000412</td>\n",
       "      <td>0.000665</td>\n",
       "      <td>0.000889</td>\n",
       "      <td>0.001450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M02</td>\n",
       "      <td>A/G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.451613</td>\n",
       "      <td>17,14</td>\n",
       "      <td>31</td>\n",
       "      <td>99</td>\n",
       "      <td>372,0,492</td>\n",
       "      <td>P1A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,14</td>\n",
       "      <td>14</td>\n",
       "      <td>42</td>\n",
       "      <td>455,42,0</td>\n",
       "      <td>P2A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,35</td>\n",
       "      <td>35.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1131,105,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A2</td>\n",
       "      <td>WDR45</td>\n",
       "      <td>chrX</td>\n",
       "      <td>48935750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M02</td>\n",
       "      <td>G/A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.517857</td>\n",
       "      <td>27,29</td>\n",
       "      <td>56</td>\n",
       "      <td>99</td>\n",
       "      <td>823,0,709</td>\n",
       "      <td>P1A2</td>\n",
       "      <td>A/A</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,29</td>\n",
       "      <td>29</td>\n",
       "      <td>86</td>\n",
       "      <td>889,86,0</td>\n",
       "      <td>P2A2</td>\n",
       "      <td>A/A</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,34</td>\n",
       "      <td>34.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1021,100,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A2</td>\n",
       "      <td>CCDC22</td>\n",
       "      <td>chrX</td>\n",
       "      <td>49103280</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>0.000660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M02</td>\n",
       "      <td>A/G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>16,24</td>\n",
       "      <td>40</td>\n",
       "      <td>99</td>\n",
       "      <td>604,0,414</td>\n",
       "      <td>P1A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,26</td>\n",
       "      <td>26</td>\n",
       "      <td>78</td>\n",
       "      <td>795,78,0</td>\n",
       "      <td>P2A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,40</td>\n",
       "      <td>40.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1145,119,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A2</td>\n",
       "      <td>STARD8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>67940194</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000976</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>0.001935</td>\n",
       "      <td>0.000924</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M02</td>\n",
       "      <td>C/T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>15,20</td>\n",
       "      <td>35</td>\n",
       "      <td>99</td>\n",
       "      <td>536,0,359</td>\n",
       "      <td>P1A2</td>\n",
       "      <td>T/T</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,23</td>\n",
       "      <td>23</td>\n",
       "      <td>69</td>\n",
       "      <td>747,69,0</td>\n",
       "      <td>P2A2</td>\n",
       "      <td>T/T</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1325,126,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A2</td>\n",
       "      <td>SLC6A8</td>\n",
       "      <td>chrX</td>\n",
       "      <td>152960226</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000255</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000316</td>\n",
       "      <td>0.002117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M02</td>\n",
       "      <td>C/G</td>\n",
       "      <td>1</td>\n",
       "      <td>0.489362</td>\n",
       "      <td>24,23</td>\n",
       "      <td>47</td>\n",
       "      <td>99</td>\n",
       "      <td>604,0,646</td>\n",
       "      <td>P1A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,37</td>\n",
       "      <td>37</td>\n",
       "      <td>99</td>\n",
       "      <td>1202,110,0</td>\n",
       "      <td>P2A2</td>\n",
       "      <td>G/G</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0,44</td>\n",
       "      <td>44.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1380,132,0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       inheritance_mode         project_id    family_id    gene chrom  \\\n",
       "0  homozygous_recessive  snapper_merck_wgs  Family-P1A2  PTCHD1  chrX   \n",
       "1  homozygous_recessive  snapper_merck_wgs  Family-P1A2   WDR45  chrX   \n",
       "2  homozygous_recessive  snapper_merck_wgs  Family-P1A2  CCDC22  chrX   \n",
       "3  homozygous_recessive  snapper_merck_wgs  Family-P1A2  STARD8  chrX   \n",
       "4  homozygous_recessive  snapper_merck_wgs  Family-P1A2  SLC6A8  chrX   \n",
       "\n",
       "         pos ref alt         rsid annotation   exac_af    1kg_af  \\\n",
       "0   23397873   A   G  rs147324438   missense  0.000412  0.000665   \n",
       "1   48935750   G   A          NaN   missense  0.000000  0.000000   \n",
       "2   49103280   A   G          NaN   missense  0.000017  0.000000   \n",
       "3   67940194   C   T          NaN   missense  0.000976  0.000225   \n",
       "4  152960226   C   G          NaN   missense  0.000255  0.000000   \n",
       "\n",
       "   exac_popmax_af  merck_wgs_3793_af  Unnamed: 14 genotype0_sample_id  \\\n",
       "0        0.000889           0.001450          NaN                 M02   \n",
       "1        0.000000           0.000660          NaN                 M02   \n",
       "2        0.000051           0.000660          NaN                 M02   \n",
       "3        0.001935           0.000924          NaN                 M02   \n",
       "4        0.000316           0.002117          NaN                 M02   \n",
       "\n",
       "  genotype0_str  genotype0_num_alt  genotype0_allele_balance genotype0_AD  \\\n",
       "0           A/G                  1                  0.451613        17,14   \n",
       "1           G/A                  1                  0.517857        27,29   \n",
       "2           A/G                  1                  0.600000        16,24   \n",
       "3           C/T                  1                  0.571429        15,20   \n",
       "4           C/G                  1                  0.489362        24,23   \n",
       "\n",
       "   genotype0_DP  genotype0_GQ genotype0_PL genotype1_sample_id genotype1_str  \\\n",
       "0            31            99    372,0,492                P1A2           G/G   \n",
       "1            56            99    823,0,709                P1A2           A/A   \n",
       "2            40            99    604,0,414                P1A2           G/G   \n",
       "3            35            99    536,0,359                P1A2           T/T   \n",
       "4            47            99    604,0,646                P1A2           G/G   \n",
       "\n",
       "   genotype1_num_alt  genotype1_allele_balance genotype1_AD  genotype1_DP  \\\n",
       "0                  2                       1.0         0,14            14   \n",
       "1                  2                       1.0         0,29            29   \n",
       "2                  2                       1.0         0,26            26   \n",
       "3                  2                       1.0         0,23            23   \n",
       "4                  2                       1.0         0,37            37   \n",
       "\n",
       "   genotype1_GQ genotype1_PL genotype2_sample_id genotype2_str  \\\n",
       "0            42     455,42,0                P2A2           G/G   \n",
       "1            86     889,86,0                P2A2           A/A   \n",
       "2            78     795,78,0                P2A2           G/G   \n",
       "3            69     747,69,0                P2A2           T/T   \n",
       "4            99   1202,110,0                P2A2           G/G   \n",
       "\n",
       "   genotype2_num_alt  genotype2_allele_balance genotype2_AD  genotype2_DP  \\\n",
       "0                2.0                       1.0         0,35          35.0   \n",
       "1                2.0                       1.0         0,34          34.0   \n",
       "2                2.0                       1.0         0,40          40.0   \n",
       "3                2.0                       1.0         0,42          42.0   \n",
       "4                2.0                       1.0         0,44          44.0   \n",
       "\n",
       "   genotype2_GQ genotype2_PL genotype3_sample_id genotype3_str  \\\n",
       "0          99.0   1131,105,0                 NaN           NaN   \n",
       "1          99.0   1021,100,0                 NaN           NaN   \n",
       "2          99.0   1145,119,0                 NaN           NaN   \n",
       "3          99.0   1325,126,0                 NaN           NaN   \n",
       "4          99.0   1380,132,0                 NaN           NaN   \n",
       "\n",
       "   genotype3_num_alt  genotype3_allele_balance genotype3_AD  genotype3_DP  \\\n",
       "0                NaN                       NaN          NaN           NaN   \n",
       "1                NaN                       NaN          NaN           NaN   \n",
       "2                NaN                       NaN          NaN           NaN   \n",
       "3                NaN                       NaN          NaN           NaN   \n",
       "4                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "   genotype3_GQ genotype3_PL genotype4_sample_id genotype4_str  \\\n",
       "0           NaN          NaN                 NaN           NaN   \n",
       "1           NaN          NaN                 NaN           NaN   \n",
       "2           NaN          NaN                 NaN           NaN   \n",
       "3           NaN          NaN                 NaN           NaN   \n",
       "4           NaN          NaN                 NaN           NaN   \n",
       "\n",
       "   genotype4_num_alt  genotype4_allele_balance genotype4_AD  genotype4_DP  \\\n",
       "0                NaN                       NaN          NaN           NaN   \n",
       "1                NaN                       NaN          NaN           NaN   \n",
       "2                NaN                       NaN          NaN           NaN   \n",
       "3                NaN                       NaN          NaN           NaN   \n",
       "4                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "   genotype4_GQ genotype4_PL genotype5_sample_id genotype5_str  \\\n",
       "0           NaN          NaN                 NaN           NaN   \n",
       "1           NaN          NaN                 NaN           NaN   \n",
       "2           NaN          NaN                 NaN           NaN   \n",
       "3           NaN          NaN                 NaN           NaN   \n",
       "4           NaN          NaN                 NaN           NaN   \n",
       "\n",
       "   genotype5_num_alt  genotype5_allele_balance genotype5_AD  genotype5_DP  \\\n",
       "0                NaN                       NaN          NaN           NaN   \n",
       "1                NaN                       NaN          NaN           NaN   \n",
       "2                NaN                       NaN          NaN           NaN   \n",
       "3                NaN                       NaN          NaN           NaN   \n",
       "4                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "   genotype5_GQ genotype5_PL  genotype6_sample_id  genotype6_str  \\\n",
       "0           NaN          NaN                  NaN            NaN   \n",
       "1           NaN          NaN                  NaN            NaN   \n",
       "2           NaN          NaN                  NaN            NaN   \n",
       "3           NaN          NaN                  NaN            NaN   \n",
       "4           NaN          NaN                  NaN            NaN   \n",
       "\n",
       "   genotype6_num_alt  genotype6_allele_balance  genotype6_AD  genotype6_DP  \\\n",
       "0                NaN                       NaN           NaN           NaN   \n",
       "1                NaN                       NaN           NaN           NaN   \n",
       "2                NaN                       NaN           NaN           NaN   \n",
       "3                NaN                       NaN           NaN           NaN   \n",
       "4                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "   genotype6_GQ  genotype6_PL  genotype7_sample_id  genotype7_str  \\\n",
       "0           NaN           NaN                  NaN            NaN   \n",
       "1           NaN           NaN                  NaN            NaN   \n",
       "2           NaN           NaN                  NaN            NaN   \n",
       "3           NaN           NaN                  NaN            NaN   \n",
       "4           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "   genotype7_num_alt  genotype7_allele_balance  genotype7_AD  genotype7_DP  \\\n",
       "0                NaN                       NaN           NaN           NaN   \n",
       "1                NaN                       NaN           NaN           NaN   \n",
       "2                NaN                       NaN           NaN           NaN   \n",
       "3                NaN                       NaN           NaN           NaN   \n",
       "4                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "   genotype7_GQ  genotype7_PL  genotype8_sample_id  genotype8_str  \\\n",
       "0           NaN           NaN                  NaN            NaN   \n",
       "1           NaN           NaN                  NaN            NaN   \n",
       "2           NaN           NaN                  NaN            NaN   \n",
       "3           NaN           NaN                  NaN            NaN   \n",
       "4           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "   genotype8_num_alt  genotype8_allele_balance  genotype8_AD  genotype8_DP  \\\n",
       "0                NaN                       NaN           NaN           NaN   \n",
       "1                NaN                       NaN           NaN           NaN   \n",
       "2                NaN                       NaN           NaN           NaN   \n",
       "3                NaN                       NaN           NaN           NaN   \n",
       "4                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "   genotype8_GQ  genotype8_PL  genotype9_sample_id  genotype9_str  \\\n",
       "0           NaN           NaN                  NaN            NaN   \n",
       "1           NaN           NaN                  NaN            NaN   \n",
       "2           NaN           NaN                  NaN            NaN   \n",
       "3           NaN           NaN                  NaN            NaN   \n",
       "4           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "   genotype9_num_alt  genotype9_allele_balance  genotype9_AD  genotype9_DP  \\\n",
       "0                NaN                       NaN           NaN           NaN   \n",
       "1                NaN                       NaN           NaN           NaN   \n",
       "2                NaN                       NaN           NaN           NaN   \n",
       "3                NaN                       NaN           NaN           NaN   \n",
       "4                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "   genotype9_GQ  genotype9_PL  \n",
       "0           NaN           NaN  \n",
       "1           NaN           NaN  \n",
       "2           NaN           NaN  \n",
       "3           NaN           NaN  \n",
       "4           NaN           NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merk.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.279096",
     "start_time": "2016-04-27T11:43:42.150543"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Family-P1A2', 'Family-P1A04', 'Family-P1A06', 'Family-CHB-005',\n",
       "       'Family-CHB-080', 'Family-CHB-099', 'Family-CHB-106',\n",
       "       'Family-CHB-116', 'Family-CHB-143', 'Family-CHB-147',\n",
       "       'Family-CHB-156', 'Family-CHB-312', 'Family-CHWA-13',\n",
       "       'Family-AKSH-001', 'Family-P1A07', 'Family-P1A08', 'Family-P1A09',\n",
       "       'Family-P1A10', 'Family-P1A16', 'Family-CHB-159', 'Family-CHB-188',\n",
       "       'Family-CHB-230', 'Family-CHB-290', 'Family-P1A12', 'Family-P1A13'], dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merk.family_id.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.393454",
     "start_time": "2016-04-27T11:43:42.281676"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(25,)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merk.family_id.unique().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.511474",
     "start_time": "2016-04-27T11:43:42.398641"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['missense', 'frameshift', 'essential_splice_site', 'inframe',\n",
       "       'nonsense'], dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merk.annotation.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.667335",
     "start_time": "2016-04-27T11:43:42.512985"
    },
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Solved?</th>\n",
       "      <th>family_name</th>\n",
       "      <th>WES/WGS</th>\n",
       "      <th>Family_size</th>\n",
       "      <th>Likely inheritance model</th>\n",
       "      <th>Research_grouping</th>\n",
       "      <th>short phenotype description</th>\n",
       "      <th>Merckanalysis top candidates</th>\n",
       "      <th>Merck comments</th>\n",
       "      <th>Snapperanalysis top candidates</th>\n",
       "      <th>Snapper Candidate descriptions</th>\n",
       "      <th>Prioritized candidate(s)</th>\n",
       "      <th>Prioritized candidate(s) description</th>\n",
       "      <th>Other Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family AKSH-001</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Consanguineous family with child with colitis ...</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-005</td>\n",
       "      <td>WGS</td>\n",
       "      <td>5</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 of 3 siblings diagnosed with CD. CHB005 at 1...</td>\n",
       "      <td>TRPM5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-080</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Proband with UC and PSC symptom onset 6 months...</td>\n",
       "      <td>KIAA1875, FARP2, CLASRP, RBBP7, ZNF81</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-099</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>CHB099 with Ulcerative colitis diagnosed at 2....</td>\n",
       "      <td>FBXL13, SMG1, MYO1F, AKT3, HECTD1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-106</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Symptom onset and diagnosed with indeterminate...</td>\n",
       "      <td>PKD1L2, NLGN3, SH3BGRL, IGSF1, IFNL3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IFNL3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-116</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Symptom onset 1 month old. Diagnosed with \"IBD...</td>\n",
       "      <td>FLT1, KRT8, IRAK1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-143</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Sister with CD diagnosed at 23 years old and s...</td>\n",
       "      <td>ADD3</td>\n",
       "      <td>CNV (cmp het)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-147</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Patient diagnosed at 2 years of age with UC, b...</td>\n",
       "      <td>C1GALT1C1, IDS, PLXNA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLXNA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-156</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Symptoms of CHB156 at about 5 months of age.</td>\n",
       "      <td>MEGF6, RNF212, PXMP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-159</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>CHB159 diagnosed with CD at 5 years of age.</td>\n",
       "      <td>BTK, ICOSLG, ZNF99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BTK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-188</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Proband has UC diagnosed at 3 years old, now s...</td>\n",
       "      <td>MTUS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-230</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Proband with UC.</td>\n",
       "      <td>PLXNA3, TBC1D25, PASD1, CELF1</td>\n",
       "      <td>CELF1 is a CNV (hmz rec deletion)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLXNA3, TBC1D25, PASD1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Proband has had a heart transplant, immune dys...</td>\n",
       "      <td>NCOA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCOA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-312</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Proband symptomatic since 2 months of age, dia...</td>\n",
       "      <td>NDUFA13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHB-321</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 siblings with UC. No other family history of...</td>\n",
       "      <td>ZNF587P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family CHWA-13</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Patient symptomatic at 6 months of age, diagno...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Only 2 individuals sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A04</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 sisters diagnosed with Crohn's disease. SM-6...</td>\n",
       "      <td>KRT78, PLEKHG4, PNPLA6,F5, UGT2A3, UGT2B1, GRIP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A06</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 siblings each diagnosed with Crohn's. SM-6BO...</td>\n",
       "      <td>IFT172, PTPRD, DHTKD1, SIRT3, NOC4L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A07</td>\n",
       "      <td>WGS</td>\n",
       "      <td>5</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>3 sibs with IBD, ulcerative colitis and indete...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A08</td>\n",
       "      <td>WGS</td>\n",
       "      <td>6</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 of triplets affects with UC, with 2 unaffect...</td>\n",
       "      <td>DNAJB4, NUTM2D, HIST1H2BD, VAV1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A09</td>\n",
       "      <td>WGS</td>\n",
       "      <td>5</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 siblings each diagnosed with CD at 3 years o...</td>\n",
       "      <td>IFI16</td>\n",
       "      <td>CNV (cmp het)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A10</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>P1A10 diagnosed with CD at 12 years old; P2A10...</td>\n",
       "      <td>MAML3, LPA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A12</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>P1A12 diagnosed with UC at 13 y/o, shortly aft...</td>\n",
       "      <td>TACC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A13</td>\n",
       "      <td>WGS</td>\n",
       "      <td>5</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Family with 4 children diagnosed with CD &lt; 18 ...</td>\n",
       "      <td>PDZD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A16</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>Siblings diagnosed with UC and CD. No family h...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Family P1A2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>recessive</td>\n",
       "      <td>Snapper</td>\n",
       "      <td>2 brothers of Greek descent diagnosed with Cro...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Father not sequenced. No candidate variants fr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Solved?      family_name WES/WGS  Family_size Likely inheritance model  \\\n",
       "0       NaN  Family AKSH-001     WGS            2       de_novo, recessive   \n",
       "1       NaN   Family CHB-005     WGS            5                recessive   \n",
       "2       NaN   Family CHB-080     WGS            3       de_novo, recessive   \n",
       "3       NaN   Family CHB-099     WGS            3       de_novo, recessive   \n",
       "4       NaN   Family CHB-106     WGS            3       de_novo, recessive   \n",
       "5       NaN   Family CHB-116     WGS            3       de_novo, recessive   \n",
       "6       NaN   Family CHB-143     WGS            4       de_novo, recessive   \n",
       "7       NaN   Family CHB-147     WGS            3       de_novo, recessive   \n",
       "8       NaN   Family CHB-156     WGS            3       de_novo, recessive   \n",
       "9       NaN   Family CHB-159     WGS            3       de_novo, recessive   \n",
       "10      NaN   Family CHB-188     WGS            3       de_novo, recessive   \n",
       "11      NaN   Family CHB-230     WGS            3       de_novo, recessive   \n",
       "12      NaN   Family CHB-290     WGS            4                recessive   \n",
       "13      NaN   Family CHB-312     WGS            3       de_novo, recessive   \n",
       "14      NaN   Family CHB-321     WGS            4                recessive   \n",
       "15      NaN   Family CHWA-13     WGS            2       de_novo, recessive   \n",
       "16      NaN     Family P1A04     WGS            4       de_novo, recessive   \n",
       "17      NaN     Family P1A06     WGS            3       de_novo, recessive   \n",
       "18      NaN     Family P1A07     WGS            5                recessive   \n",
       "19      NaN     Family P1A08     WGS            6                recessive   \n",
       "20      NaN     Family P1A09     WGS            5                recessive   \n",
       "21      NaN     Family P1A10     WGS            4                recessive   \n",
       "22      NaN     Family P1A12     WGS            4                recessive   \n",
       "23      NaN     Family P1A13     WGS            5                recessive   \n",
       "24      NaN     Family P1A16     WGS            4                recessive   \n",
       "25      NaN      Family P1A2     WGS            3                recessive   \n",
       "\n",
       "   Research_grouping                        short phenotype description  \\\n",
       "0            Snapper  Consanguineous family with child with colitis ...   \n",
       "1            Snapper  2 of 3 siblings diagnosed with CD. CHB005 at 1...   \n",
       "2            Snapper  Proband with UC and PSC symptom onset 6 months...   \n",
       "3            Snapper  CHB099 with Ulcerative colitis diagnosed at 2....   \n",
       "4            Snapper  Symptom onset and diagnosed with indeterminate...   \n",
       "5            Snapper  Symptom onset 1 month old. Diagnosed with \"IBD...   \n",
       "6            Snapper  Sister with CD diagnosed at 23 years old and s...   \n",
       "7            Snapper  Patient diagnosed at 2 years of age with UC, b...   \n",
       "8            Snapper       Symptoms of CHB156 at about 5 months of age.   \n",
       "9            Snapper        CHB159 diagnosed with CD at 5 years of age.   \n",
       "10           Snapper  Proband has UC diagnosed at 3 years old, now s...   \n",
       "11           Snapper                                   Proband with UC.   \n",
       "12           Snapper  Proband has had a heart transplant, immune dys...   \n",
       "13           Snapper  Proband symptomatic since 2 months of age, dia...   \n",
       "14           Snapper  2 siblings with UC. No other family history of...   \n",
       "15           Snapper  Patient symptomatic at 6 months of age, diagno...   \n",
       "16           Snapper  2 sisters diagnosed with Crohn's disease. SM-6...   \n",
       "17           Snapper  2 siblings each diagnosed with Crohn's. SM-6BO...   \n",
       "18           Snapper  3 sibs with IBD, ulcerative colitis and indete...   \n",
       "19           Snapper  2 of triplets affects with UC, with 2 unaffect...   \n",
       "20           Snapper  2 siblings each diagnosed with CD at 3 years o...   \n",
       "21           Snapper  P1A10 diagnosed with CD at 12 years old; P2A10...   \n",
       "22           Snapper  P1A12 diagnosed with UC at 13 y/o, shortly aft...   \n",
       "23           Snapper  Family with 4 children diagnosed with CD < 18 ...   \n",
       "24           Snapper  Siblings diagnosed with UC and CD. No family h...   \n",
       "25           Snapper  2 brothers of Greek descent diagnosed with Cro...   \n",
       "\n",
       "                        Merckanalysis top candidates  \\\n",
       "0                                              DOCK8   \n",
       "1                                              TRPM5   \n",
       "2              KIAA1875, FARP2, CLASRP, RBBP7, ZNF81   \n",
       "3                  FBXL13, SMG1, MYO1F, AKT3, HECTD1   \n",
       "4               PKD1L2, NLGN3, SH3BGRL, IGSF1, IFNL3   \n",
       "5                                  FLT1, KRT8, IRAK1   \n",
       "6                                               ADD3   \n",
       "7                             C1GALT1C1, IDS, PLXNA3   \n",
       "8                               MEGF6, RNF212, PXMP2   \n",
       "9                                 BTK, ICOSLG, ZNF99   \n",
       "10                                             MTUS1   \n",
       "11                     PLXNA3, TBC1D25, PASD1, CELF1   \n",
       "12                                             NCOA3   \n",
       "13                                           NDUFA13   \n",
       "14                                          ZNF587P1   \n",
       "15                                               NaN   \n",
       "16  KRT78, PLEKHG4, PNPLA6,F5, UGT2A3, UGT2B1, GRIP1   \n",
       "17               IFT172, PTPRD, DHTKD1, SIRT3, NOC4L   \n",
       "18                                               NaN   \n",
       "19                   DNAJB4, NUTM2D, HIST1H2BD, VAV1   \n",
       "20                                             IFI16   \n",
       "21                                        MAML3, LPA   \n",
       "22                                             TACC2   \n",
       "23                                             PDZD2   \n",
       "24                                               NaN   \n",
       "25                                               NaN   \n",
       "\n",
       "                                       Merck comments  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3                                                 NaN   \n",
       "4                                                 NaN   \n",
       "5                                                 NaN   \n",
       "6                                       CNV (cmp het)   \n",
       "7                                                 NaN   \n",
       "8                                                 NaN   \n",
       "9                                                 NaN   \n",
       "10                                                NaN   \n",
       "11                  CELF1 is a CNV (hmz rec deletion)   \n",
       "12                                                NaN   \n",
       "13                                                NaN   \n",
       "14                                                NaN   \n",
       "15                       Only 2 individuals sequenced   \n",
       "16                                                NaN   \n",
       "17                                                NaN   \n",
       "18                                                NaN   \n",
       "19                                                NaN   \n",
       "20                                      CNV (cmp het)   \n",
       "21                                                NaN   \n",
       "22                                                NaN   \n",
       "23                                                NaN   \n",
       "24                                                NaN   \n",
       "25  Father not sequenced. No candidate variants fr...   \n",
       "\n",
       "    Snapperanalysis top candidates  Snapper Candidate descriptions  \\\n",
       "0                              NaN                             NaN   \n",
       "1                              NaN                             NaN   \n",
       "2                              NaN                             NaN   \n",
       "3                              NaN                             NaN   \n",
       "4                              NaN                             NaN   \n",
       "5                              NaN                             NaN   \n",
       "6                              NaN                             NaN   \n",
       "7                              NaN                             NaN   \n",
       "8                              NaN                             NaN   \n",
       "9                              NaN                             NaN   \n",
       "10                             NaN                             NaN   \n",
       "11                             NaN                             NaN   \n",
       "12                             NaN                             NaN   \n",
       "13                             NaN                             NaN   \n",
       "14                             NaN                             NaN   \n",
       "15                             NaN                             NaN   \n",
       "16                             NaN                             NaN   \n",
       "17                             NaN                             NaN   \n",
       "18                             NaN                             NaN   \n",
       "19                             NaN                             NaN   \n",
       "20                             NaN                             NaN   \n",
       "21                             NaN                             NaN   \n",
       "22                             NaN                             NaN   \n",
       "23                             NaN                             NaN   \n",
       "24                             NaN                             NaN   \n",
       "25                             NaN                             NaN   \n",
       "\n",
       "   Prioritized candidate(s)  Prioritized candidate(s) description  \\\n",
       "0                     DOCK8                                   NaN   \n",
       "1                       NaN                                   NaN   \n",
       "2                       NaN                                   NaN   \n",
       "3                       NaN                                   NaN   \n",
       "4                     IFNL3                                   NaN   \n",
       "5                       NaN                                   NaN   \n",
       "6                       NaN                                   NaN   \n",
       "7                    PLXNA3                                   NaN   \n",
       "8                       NaN                                   NaN   \n",
       "9                       BTK                                   NaN   \n",
       "10                      NaN                                   NaN   \n",
       "11   PLXNA3, TBC1D25, PASD1                                   NaN   \n",
       "12                    NCOA3                                   NaN   \n",
       "13                      NaN                                   NaN   \n",
       "14                      NaN                                   NaN   \n",
       "15                      NaN                                   NaN   \n",
       "16                      NaN                                   NaN   \n",
       "17                      NaN                                   NaN   \n",
       "18                      NaN                                   NaN   \n",
       "19                      NaN                                   NaN   \n",
       "20                      NaN                                   NaN   \n",
       "21                      NaN                                   NaN   \n",
       "22                      NaN                                   NaN   \n",
       "23                      NaN                                   NaN   \n",
       "24                      NaN                                   NaN   \n",
       "25                      NaN                                   NaN   \n",
       "\n",
       "    Other Comments  \n",
       "0              NaN  \n",
       "1              NaN  \n",
       "2              NaN  \n",
       "3              NaN  \n",
       "4              NaN  \n",
       "5              NaN  \n",
       "6              NaN  \n",
       "7              NaN  \n",
       "8              NaN  \n",
       "9              NaN  \n",
       "10             NaN  \n",
       "11             NaN  \n",
       "12             NaN  \n",
       "13             NaN  \n",
       "14             NaN  \n",
       "15             NaN  \n",
       "16             NaN  \n",
       "17             NaN  \n",
       "18             NaN  \n",
       "19             NaN  \n",
       "20             NaN  \n",
       "21             NaN  \n",
       "22             NaN  \n",
       "23             NaN  \n",
       "24             NaN  \n",
       "25             NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.752195",
     "start_time": "2016-04-27T11:43:42.669271"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Family AKSH-001', 'Family CHB-005', 'Family CHB-080',\n",
       "       'Family CHB-099', 'Family CHB-106', 'Family CHB-116',\n",
       "       'Family CHB-143', 'Family CHB-147', 'Family CHB-156',\n",
       "       'Family CHB-159', 'Family CHB-188', 'Family CHB-230',\n",
       "       'Family CHB-290', 'Family CHB-312', 'Family CHB-321',\n",
       "       'Family CHWA-13', 'Family P1A04', 'Family P1A06', 'Family P1A07',\n",
       "       'Family P1A08', 'Family P1A09', 'Family P1A10', 'Family P1A12',\n",
       "       'Family P1A13', 'Family P1A16', 'Family P1A2'], dtype=object)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cand.family_name.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:42.854625",
     "start_time": "2016-04-27T11:43:42.754464"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(26,)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cand.family_name.unique().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table manipulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:45:58.189493",
     "start_time": "2016-04-27T11:45:58.143931"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def top_cands_prune_columns(df):\n",
    "    \"\"\"Return pruned table with one row per top candidate.\"\"\"\n",
    "    col_map = {\"family_name\": \"family\",\n",
    "               \"WES/WGS\": \"seq_type\",\n",
    "               \"Family_size\": \"fam_size\",\n",
    "               \"Likely inheritance model\": \"inherit_mode\",\n",
    "               \"short phenotype description\": \"phenotype\",\n",
    "               \"Merckanalysis top candidates\": \"top_candidates\",\n",
    "              }\n",
    "    \n",
    "    keep = [\"family\",\"seq_type\",\"fam_size\",\"inherit_mode\",\"top_candidates\",\"phenotype\"]\n",
    "    \n",
    "    # prune columns and clean some row values\n",
    "    pruned_table = df.rename(columns=col_map)\n",
    "    pruned_table = pruned_table[keep].copy().dropna()\n",
    "    pruned_table[\"top_candidates\"] = pruned_table[\"top_candidates\"].apply(lambda i: i.upper().strip().replace(' ','').split(','))\n",
    "    pruned_table[\"family\"] = pruned_table[\"family\"].apply(lambda i: i.upper().replace(\"FAMILY\",\"\").strip())\n",
    "    \n",
    "    return explode_col(df=pruned_table, idx_col=\"family\", exp_col=\"top_candidates\")[keep]\n",
    "\n",
    "def explode_col(df, idx_col, exp_col):\n",
    "    \"\"\"Return dataframe where `exp_col` is exploded and then merged back into `df` using `idx_col`.\n",
    "    \n",
    "    `exp_col` must contain existing lists.\n",
    "    `idx_col` should be suitible for use as a primary key (unique).\n",
    "    \"\"\"\n",
    "    keep = [col for col in df.columns.values if col != exp_col]\n",
    "    \n",
    "    # explode the top_candidates row lists into new dataframe\n",
    "    exploded = df[[idx_col,exp_col]].set_index(idx_col)\n",
    "    exploded = exploded.to_dict()[exp_col]\n",
    "    \n",
    "    idx_value_pairs = []\n",
    "    for idx, values in exploded.items():\n",
    "        idx_value_pairs.extend([[idx, value] for value in values])\n",
    "\n",
    "    \n",
    "    exploded = pd.DataFrame(idx_value_pairs, columns=[idx_col,exp_col]).set_index(idx_col)\n",
    "    \n",
    "    # join the exploded df with the rest of the data we want to keep\n",
    "    remerged = pd.merge(left=df[keep].set_index(idx_col), right=exploded,\n",
    "                      how=\"inner\", \n",
    "                      on=None, left_on=None, right_on=None, \n",
    "                      left_index=True, right_index=True,)\n",
    "    \n",
    "    return remerged.reset_index(idx_col).copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Working with the graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.050207",
     "start_time": "2016-04-27T11:43:42.945951"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_nodes_by_attr_values(g, attr_name, attr_values):\n",
    "    \"\"\"Return generator of nodes whose values for `attr_name` are in `attr_values`.\"\"\"\n",
    "    attr = attr_name\n",
    "    vals = set(attr_values)\n",
    "    \n",
    "    return (n for n,d in g.nodes_iter(data=True) if d[attr] in vals)\n",
    "\n",
    "def check_nodes_for_paths(g, nodes):\n",
    "    \"\"\"Return dictionary keyed with directional node tuples and valued with the shortest path if any between purmuations of the `nodes` list.\"\"\"\n",
    "    all_shortest = nx.all_pairs_shortest_path(g)\n",
    "    node_pairs = itertools.permutations(nodes, 2)\n",
    "    \n",
    "    has_path = {}\n",
    "    for source, target in node_pairs:\n",
    "        try:\n",
    "            path = all_shortest[source][target]\n",
    "            has_path[(source,target)] = path\n",
    "        except KeyError:\n",
    "            has_path[(source,target)] = None\n",
    "            \n",
    "    return has_path\n",
    "\n",
    "def get_interactions_from_path(g, path):\n",
    "    \"\"\"Return interleaved list of nodes and interactions of the path provided.\"\"\"\n",
    "    sources = path[:-1]\n",
    "    targets = path[1:]\n",
    "    \n",
    "    interactions = []\n",
    "    for s, t in zip(sources, targets):\n",
    "        \n",
    "        if g.node[s]['name'] not in interactions:\n",
    "            interactions.append(g.node[s]['name'])\n",
    "        \n",
    "        interactions.append(g[s][t]['interaction'])\n",
    "        interactions.append(g.node[t]['name'])\n",
    "    \n",
    "    return interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prune candidates table to just some specific info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:46:02.405046",
     "start_time": "2016-04-27T11:46:02.320361"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "topc = top_cands_prune_columns(cand)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:58:20.807192",
     "start_time": "2016-04-27T11:58:20.790530"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>family</th>\n",
       "      <th>seq_type</th>\n",
       "      <th>fam_size</th>\n",
       "      <th>inherit_mode</th>\n",
       "      <th>top_candidates</th>\n",
       "      <th>phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKSH-001</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>Consanguineous family with child with colitis ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHB-005</td>\n",
       "      <td>WGS</td>\n",
       "      <td>5</td>\n",
       "      <td>recessive</td>\n",
       "      <td>TRPM5</td>\n",
       "      <td>2 of 3 siblings diagnosed with CD. CHB005 at 1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CHB-080</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>KIAA1875</td>\n",
       "      <td>Proband with UC and PSC symptom onset 6 months...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CHB-080</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>FARP2</td>\n",
       "      <td>Proband with UC and PSC symptom onset 6 months...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHB-080</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>CLASRP</td>\n",
       "      <td>Proband with UC and PSC symptom onset 6 months...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     family seq_type  fam_size        inherit_mode top_candidates  \\\n",
       "0  AKSH-001      WGS         2  de_novo, recessive          DOCK8   \n",
       "1   CHB-005      WGS         5           recessive          TRPM5   \n",
       "2   CHB-080      WGS         3  de_novo, recessive       KIAA1875   \n",
       "3   CHB-080      WGS         3  de_novo, recessive          FARP2   \n",
       "4   CHB-080      WGS         3  de_novo, recessive         CLASRP   \n",
       "\n",
       "                                           phenotype  \n",
       "0  Consanguineous family with child with colitis ...  \n",
       "1  2 of 3 siblings diagnosed with CD. CHB005 at 1...  \n",
       "2  Proband with UC and PSC symptom onset 6 months...  \n",
       "3  Proband with UC and PSC symptom onset 6 months...  \n",
       "4  Proband with UC and PSC symptom onset 6 months...  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.366799",
     "start_time": "2016-04-27T11:43:43.302627"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(59, 6)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topc.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.521449",
     "start_time": "2016-04-27T11:43:43.368248"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(58,)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topc.top_candidates.unique().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## List of top candidate genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.733733",
     "start_time": "2016-04-27T11:43:43.523471"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "top_genes_uniq = list(topc.top_candidates.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.844401",
     "start_time": "2016-04-27T11:43:43.736166"
    },
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['DOCK8',\n",
       " 'TRPM5',\n",
       " 'KIAA1875',\n",
       " 'FARP2',\n",
       " 'CLASRP',\n",
       " 'RBBP7',\n",
       " 'ZNF81',\n",
       " 'FBXL13',\n",
       " 'SMG1',\n",
       " 'MYO1F',\n",
       " 'AKT3',\n",
       " 'HECTD1',\n",
       " 'PKD1L2',\n",
       " 'NLGN3',\n",
       " 'SH3BGRL',\n",
       " 'IGSF1',\n",
       " 'IFNL3',\n",
       " 'FLT1',\n",
       " 'KRT8',\n",
       " 'IRAK1',\n",
       " 'ADD3',\n",
       " 'C1GALT1C1',\n",
       " 'IDS',\n",
       " 'PLXNA3',\n",
       " 'MEGF6',\n",
       " 'RNF212',\n",
       " 'PXMP2',\n",
       " 'BTK',\n",
       " 'ICOSLG',\n",
       " 'ZNF99',\n",
       " 'MTUS1',\n",
       " 'TBC1D25',\n",
       " 'PASD1',\n",
       " 'CELF1',\n",
       " 'NCOA3',\n",
       " 'NDUFA13',\n",
       " 'ZNF587P1',\n",
       " 'KRT78',\n",
       " 'PLEKHG4',\n",
       " 'PNPLA6',\n",
       " 'F5',\n",
       " 'UGT2A3',\n",
       " 'UGT2B1',\n",
       " 'GRIP1',\n",
       " 'IFT172',\n",
       " 'PTPRD',\n",
       " 'DHTKD1',\n",
       " 'SIRT3',\n",
       " 'NOC4L',\n",
       " 'DNAJB4',\n",
       " 'NUTM2D',\n",
       " 'HIST1H2BD',\n",
       " 'VAV1',\n",
       " 'IFI16',\n",
       " 'MAML3',\n",
       " 'LPA',\n",
       " 'TACC2',\n",
       " 'PDZD2']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_genes_uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Which of the top_candidate genes can we find in the IBD graph?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:43.926238",
     "start_time": "2016-04-27T11:43:43.846244"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "top_genes_ibd = list(get_nodes_by_attr_values(ibd, attr_name='name', attr_values=top_genes_uniq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:44.048903",
     "start_time": "2016-04-27T11:43:43.928733"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['4517', '1580', '1312', '6047']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_genes_ibd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:44.125367",
     "start_time": "2016-04-27T11:43:44.051117"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4517: BTK\n",
      "1580: VAV1\n",
      "1312: DOCK8\n",
      "6047: MYO1F\n"
     ]
    }
   ],
   "source": [
    "for n in top_genes_ibd:\n",
    "    print(\"{id}: {name}\".format(id=n, name=ibd.node[n]['name']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Can we find paths between any of the genes? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:44.239501",
     "start_time": "2016-04-27T11:43:44.127265"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('6047', '1580'): [('MYO1F', '6047'), ('BIN2', '1880'), ('LYN', '4247'), ('LCP2', '1579'), ('VAV1', '1580')]\n"
     ]
    }
   ],
   "source": [
    "has_path = check_nodes_for_paths(g=ibd, nodes=top_genes_ibd)\n",
    "for pair, path in has_path.items():\n",
    "    if path is not None:\n",
    "        print(\"{pair}: {path}\".format(pair=pair, path=[(ibd.node[n][\"name\"],n) for n in path]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:43:44.327336",
     "start_time": "2016-04-27T11:43:44.241464"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'pp'"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ibd['1579']['1580']['interaction']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:44:07.474950",
     "start_time": "2016-04-27T11:44:07.470927"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "path = has_path[('6047', '1580')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T11:44:08.023526",
     "start_time": "2016-04-27T11:44:08.018566"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MYO1F', 'pp', 'BIN2', 'pp', 'LYN', 'pp', 'LCP2', 'pp', 'VAV1']"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_interactions_from_path(g=ibd, path=path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Lets compare the symptoms from subjects with the SNPs in the genes we found that have paths linking them?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T12:01:10.170136",
     "start_time": "2016-04-27T12:01:10.156939"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>family</th>\n",
       "      <th>seq_type</th>\n",
       "      <th>fam_size</th>\n",
       "      <th>inherit_mode</th>\n",
       "      <th>top_candidates</th>\n",
       "      <th>phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CHB-099</td>\n",
       "      <td>WGS</td>\n",
       "      <td>3</td>\n",
       "      <td>de_novo, recessive</td>\n",
       "      <td>MYO1F</td>\n",
       "      <td>CHB099 with Ulcerative colitis diagnosed at 2.5 yrs old- colectomy at 12 yrs old.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>P1A08</td>\n",
       "      <td>WGS</td>\n",
       "      <td>6</td>\n",
       "      <td>recessive</td>\n",
       "      <td>VAV1</td>\n",
       "      <td>2 of triplets affects with UC, with 2 unaffected siblings and no other family history of IBD.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     family seq_type  fam_size        inherit_mode top_candidates  \\\n",
       "9   CHB-099      WGS         3  de_novo, recessive          MYO1F   \n",
       "53    P1A08      WGS         6           recessive           VAV1   \n",
       "\n",
       "                                                                                        phenotype  \n",
       "9               CHB099 with Ulcerative colitis diagnosed at 2.5 yrs old- colectomy at 12 yrs old.  \n",
       "53  2 of triplets affects with UC, with 2 unaffected siblings and no other family history of IBD.  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topc[topc[\"top_candidates\"].apply(lambda i: i in ['MYO1F','VAV1'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Any familes have SNPs in any of the other genes in that path?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2016-04-27T12:05:43.736781",
     "start_time": "2016-04-27T12:05:43.571345"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inheritance_mode</th>\n",
       "      <th>project_id</th>\n",
       "      <th>family_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>rsid</th>\n",
       "      <th>annotation</th>\n",
       "      <th>exac_af</th>\n",
       "      <th>1kg_af</th>\n",
       "      <th>exac_popmax_af</th>\n",
       "      <th>merck_wgs_3793_af</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "      <th>genotype0_sample_id</th>\n",
       "      <th>genotype0_str</th>\n",
       "      <th>genotype0_num_alt</th>\n",
       "      <th>genotype0_allele_balance</th>\n",
       "      <th>genotype0_AD</th>\n",
       "      <th>genotype0_DP</th>\n",
       "      <th>genotype0_GQ</th>\n",
       "      <th>genotype0_PL</th>\n",
       "      <th>genotype1_sample_id</th>\n",
       "      <th>genotype1_str</th>\n",
       "      <th>genotype1_num_alt</th>\n",
       "      <th>genotype1_allele_balance</th>\n",
       "      <th>genotype1_AD</th>\n",
       "      <th>genotype1_DP</th>\n",
       "      <th>genotype1_GQ</th>\n",
       "      <th>genotype1_PL</th>\n",
       "      <th>genotype2_sample_id</th>\n",
       "      <th>genotype2_str</th>\n",
       "      <th>genotype2_num_alt</th>\n",
       "      <th>genotype2_allele_balance</th>\n",
       "      <th>genotype2_AD</th>\n",
       "      <th>genotype2_DP</th>\n",
       "      <th>genotype2_GQ</th>\n",
       "      <th>genotype2_PL</th>\n",
       "      <th>genotype3_sample_id</th>\n",
       "      <th>genotype3_str</th>\n",
       "      <th>genotype3_num_alt</th>\n",
       "      <th>genotype3_allele_balance</th>\n",
       "      <th>genotype3_AD</th>\n",
       "      <th>genotype3_DP</th>\n",
       "      <th>genotype3_GQ</th>\n",
       "      <th>genotype3_PL</th>\n",
       "      <th>genotype4_sample_id</th>\n",
       "      <th>genotype4_str</th>\n",
       "      <th>genotype4_num_alt</th>\n",
       "      <th>genotype4_allele_balance</th>\n",
       "      <th>genotype4_AD</th>\n",
       "      <th>genotype4_DP</th>\n",
       "      <th>genotype4_GQ</th>\n",
       "      <th>genotype4_PL</th>\n",
       "      <th>genotype5_sample_id</th>\n",
       "      <th>genotype5_str</th>\n",
       "      <th>genotype5_num_alt</th>\n",
       "      <th>genotype5_allele_balance</th>\n",
       "      <th>genotype5_AD</th>\n",
       "      <th>genotype5_DP</th>\n",
       "      <th>genotype5_GQ</th>\n",
       "      <th>genotype5_PL</th>\n",
       "      <th>genotype6_sample_id</th>\n",
       "      <th>genotype6_str</th>\n",
       "      <th>genotype6_num_alt</th>\n",
       "      <th>genotype6_allele_balance</th>\n",
       "      <th>genotype6_AD</th>\n",
       "      <th>genotype6_DP</th>\n",
       "      <th>genotype6_GQ</th>\n",
       "      <th>genotype6_PL</th>\n",
       "      <th>genotype7_sample_id</th>\n",
       "      <th>genotype7_str</th>\n",
       "      <th>genotype7_num_alt</th>\n",
       "      <th>genotype7_allele_balance</th>\n",
       "      <th>genotype7_AD</th>\n",
       "      <th>genotype7_DP</th>\n",
       "      <th>genotype7_GQ</th>\n",
       "      <th>genotype7_PL</th>\n",
       "      <th>genotype8_sample_id</th>\n",
       "      <th>genotype8_str</th>\n",
       "      <th>genotype8_num_alt</th>\n",
       "      <th>genotype8_allele_balance</th>\n",
       "      <th>genotype8_AD</th>\n",
       "      <th>genotype8_DP</th>\n",
       "      <th>genotype8_GQ</th>\n",
       "      <th>genotype8_PL</th>\n",
       "      <th>genotype9_sample_id</th>\n",
       "      <th>genotype9_str</th>\n",
       "      <th>genotype9_num_alt</th>\n",
       "      <th>genotype9_allele_balance</th>\n",
       "      <th>genotype9_AD</th>\n",
       "      <th>genotype9_DP</th>\n",
       "      <th>genotype9_GQ</th>\n",
       "      <th>genotype9_PL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>homozygous_recessive</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-CHB-099</td>\n",
       "      <td>MYO1F</td>\n",
       "      <td>chr19</td>\n",
       "      <td>8601510</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000630</td>\n",
       "      <td>0.000660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHB-099</td>\n",
       "      <td>T/T</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0,39</td>\n",
       "      <td>39</td>\n",
       "      <td>99</td>\n",
       "      <td>1163,115,0</td>\n",
       "      <td>CHB-100</td>\n",
       "      <td>C/T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.625</td>\n",
       "      <td>9,15</td>\n",
       "      <td>24</td>\n",
       "      <td>99</td>\n",
       "      <td>344,0,166</td>\n",
       "      <td>CHB-101</td>\n",
       "      <td>C/T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.631579</td>\n",
       "      <td>7,12</td>\n",
       "      <td>19.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>281,0,173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>dominant</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A08</td>\n",
       "      <td>VAV1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6837060</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000264</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>28,0</td>\n",
       "      <td>28</td>\n",
       "      <td>69</td>\n",
       "      <td>0,69,892</td>\n",
       "      <td>M08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>25,0</td>\n",
       "      <td>25</td>\n",
       "      <td>60</td>\n",
       "      <td>0,60,796</td>\n",
       "      <td>P1A08</td>\n",
       "      <td>A/C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.473684</td>\n",
       "      <td>30,27</td>\n",
       "      <td>57.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>703,0,831</td>\n",
       "      <td>P2A08</td>\n",
       "      <td>A/C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>24,24</td>\n",
       "      <td>48.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>633,0,632</td>\n",
       "      <td>S3A08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26,0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>0,68,854</td>\n",
       "      <td>S4A08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28,0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0,70,898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>dominant</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-CHWA-13</td>\n",
       "      <td>VAV1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6848019</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>0.000132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHWA-13</td>\n",
       "      <td>C/T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.581081</td>\n",
       "      <td>31,43</td>\n",
       "      <td>74</td>\n",
       "      <td>99</td>\n",
       "      <td>1248,0,739</td>\n",
       "      <td>CHWA-14</td>\n",
       "      <td>C/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>40,0</td>\n",
       "      <td>40</td>\n",
       "      <td>98</td>\n",
       "      <td>0,98,1262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>de_novo</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-P1A08</td>\n",
       "      <td>VAV1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6837060</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000264</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>28,0</td>\n",
       "      <td>28</td>\n",
       "      <td>69</td>\n",
       "      <td>0,69,892</td>\n",
       "      <td>M08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>25,0</td>\n",
       "      <td>25</td>\n",
       "      <td>60</td>\n",
       "      <td>0,60,796</td>\n",
       "      <td>P1A08</td>\n",
       "      <td>A/C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.473684</td>\n",
       "      <td>30,27</td>\n",
       "      <td>57.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>703,0,831</td>\n",
       "      <td>P2A08</td>\n",
       "      <td>A/C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>24,24</td>\n",
       "      <td>48.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>633,0,632</td>\n",
       "      <td>S3A08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26,0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>0,68,854</td>\n",
       "      <td>S4A08</td>\n",
       "      <td>A/A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28,0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0,70,898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1684</th>\n",
       "      <td>de_novo</td>\n",
       "      <td>snapper_merck_wgs</td>\n",
       "      <td>Family-CHWA-13</td>\n",
       "      <td>VAV1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6848019</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>0.000132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHWA-13</td>\n",
       "      <td>C/T</td>\n",
       "      <td>1</td>\n",
       "      <td>0.581081</td>\n",
       "      <td>31,43</td>\n",
       "      <td>74</td>\n",
       "      <td>99</td>\n",
       "      <td>1248,0,739</td>\n",
       "      <td>CHWA-14</td>\n",
       "      <td>C/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>40,0</td>\n",
       "      <td>40</td>\n",
       "      <td>98</td>\n",
       "      <td>0,98,1262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          inheritance_mode         project_id       family_id   gene  chrom  \\\n",
       "14    homozygous_recessive  snapper_merck_wgs  Family-CHB-099  MYO1F  chr19   \n",
       "187               dominant  snapper_merck_wgs    Family-P1A08   VAV1  chr19   \n",
       "721               dominant  snapper_merck_wgs  Family-CHWA-13   VAV1  chr19   \n",
       "1176               de_novo  snapper_merck_wgs    Family-P1A08   VAV1  chr19   \n",
       "1684               de_novo  snapper_merck_wgs  Family-CHWA-13   VAV1  chr19   \n",
       "\n",
       "          pos ref alt rsid annotation   exac_af  1kg_af  exac_popmax_af  \\\n",
       "14    8601510   C   T  NaN   missense  0.000315     0.0        0.000630   \n",
       "187   6837060   A   C  NaN   missense  0.000000     0.0        0.000000   \n",
       "721   6848019   C   T  NaN   missense  0.000115     0.0        0.000199   \n",
       "1176  6837060   A   C  NaN   missense  0.000000     0.0        0.000000   \n",
       "1684  6848019   C   T  NaN   missense  0.000115     0.0        0.000199   \n",
       "\n",
       "      merck_wgs_3793_af  Unnamed: 14 genotype0_sample_id genotype0_str  \\\n",
       "14             0.000660          NaN             CHB-099           T/T   \n",
       "187            0.000264          NaN                 F08           A/A   \n",
       "721            0.000132          NaN             CHWA-13           C/T   \n",
       "1176           0.000264          NaN                 F08           A/A   \n",
       "1684           0.000132          NaN             CHWA-13           C/T   \n",
       "\n",
       "      genotype0_num_alt  genotype0_allele_balance genotype0_AD  genotype0_DP  \\\n",
       "14                    2                  1.000000         0,39            39   \n",
       "187                   0                  0.000000         28,0            28   \n",
       "721                   1                  0.581081        31,43            74   \n",
       "1176                  0                  0.000000         28,0            28   \n",
       "1684                  1                  0.581081        31,43            74   \n",
       "\n",
       "      genotype0_GQ genotype0_PL genotype1_sample_id genotype1_str  \\\n",
       "14              99   1163,115,0             CHB-100           C/T   \n",
       "187             69     0,69,892                 M08           A/A   \n",
       "721             99   1248,0,739             CHWA-14           C/C   \n",
       "1176            69     0,69,892                 M08           A/A   \n",
       "1684            99   1248,0,739             CHWA-14           C/C   \n",
       "\n",
       "      genotype1_num_alt  genotype1_allele_balance genotype1_AD  genotype1_DP  \\\n",
       "14                    1                     0.625         9,15            24   \n",
       "187                   0                     0.000         25,0            25   \n",
       "721                   0                     0.000         40,0            40   \n",
       "1176                  0                     0.000         25,0            25   \n",
       "1684                  0                     0.000         40,0            40   \n",
       "\n",
       "      genotype1_GQ genotype1_PL genotype2_sample_id genotype2_str  \\\n",
       "14              99    344,0,166             CHB-101           C/T   \n",
       "187             60     0,60,796               P1A08           A/C   \n",
       "721             98    0,98,1262                 NaN           NaN   \n",
       "1176            60     0,60,796               P1A08           A/C   \n",
       "1684            98    0,98,1262                 NaN           NaN   \n",
       "\n",
       "      genotype2_num_alt  genotype2_allele_balance genotype2_AD  genotype2_DP  \\\n",
       "14                  1.0                  0.631579         7,12          19.0   \n",
       "187                 1.0                  0.473684        30,27          57.0   \n",
       "721                 NaN                       NaN          NaN           NaN   \n",
       "1176                1.0                  0.473684        30,27          57.0   \n",
       "1684                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "      genotype2_GQ genotype2_PL genotype3_sample_id genotype3_str  \\\n",
       "14            99.0    281,0,173                 NaN           NaN   \n",
       "187           99.0    703,0,831               P2A08           A/C   \n",
       "721            NaN          NaN                 NaN           NaN   \n",
       "1176          99.0    703,0,831               P2A08           A/C   \n",
       "1684           NaN          NaN                 NaN           NaN   \n",
       "\n",
       "      genotype3_num_alt  genotype3_allele_balance genotype3_AD  genotype3_DP  \\\n",
       "14                  NaN                       NaN          NaN           NaN   \n",
       "187                 1.0                       0.5        24,24          48.0   \n",
       "721                 NaN                       NaN          NaN           NaN   \n",
       "1176                1.0                       0.5        24,24          48.0   \n",
       "1684                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "      genotype3_GQ genotype3_PL genotype4_sample_id genotype4_str  \\\n",
       "14             NaN          NaN                 NaN           NaN   \n",
       "187           99.0    633,0,632               S3A08           A/A   \n",
       "721            NaN          NaN                 NaN           NaN   \n",
       "1176          99.0    633,0,632               S3A08           A/A   \n",
       "1684           NaN          NaN                 NaN           NaN   \n",
       "\n",
       "      genotype4_num_alt  genotype4_allele_balance genotype4_AD  genotype4_DP  \\\n",
       "14                  NaN                       NaN          NaN           NaN   \n",
       "187                 0.0                       0.0         26,0          26.0   \n",
       "721                 NaN                       NaN          NaN           NaN   \n",
       "1176                0.0                       0.0         26,0          26.0   \n",
       "1684                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "      genotype4_GQ genotype4_PL genotype5_sample_id genotype5_str  \\\n",
       "14             NaN          NaN                 NaN           NaN   \n",
       "187           68.0     0,68,854               S4A08           A/A   \n",
       "721            NaN          NaN                 NaN           NaN   \n",
       "1176          68.0     0,68,854               S4A08           A/A   \n",
       "1684           NaN          NaN                 NaN           NaN   \n",
       "\n",
       "      genotype5_num_alt  genotype5_allele_balance genotype5_AD  genotype5_DP  \\\n",
       "14                  NaN                       NaN          NaN           NaN   \n",
       "187                 0.0                       0.0         28,0          28.0   \n",
       "721                 NaN                       NaN          NaN           NaN   \n",
       "1176                0.0                       0.0         28,0          28.0   \n",
       "1684                NaN                       NaN          NaN           NaN   \n",
       "\n",
       "      genotype5_GQ genotype5_PL  genotype6_sample_id  genotype6_str  \\\n",
       "14             NaN          NaN                  NaN            NaN   \n",
       "187           70.0     0,70,898                  NaN            NaN   \n",
       "721            NaN          NaN                  NaN            NaN   \n",
       "1176          70.0     0,70,898                  NaN            NaN   \n",
       "1684           NaN          NaN                  NaN            NaN   \n",
       "\n",
       "      genotype6_num_alt  genotype6_allele_balance  genotype6_AD  genotype6_DP  \\\n",
       "14                  NaN                       NaN           NaN           NaN   \n",
       "187                 NaN                       NaN           NaN           NaN   \n",
       "721                 NaN                       NaN           NaN           NaN   \n",
       "1176                NaN                       NaN           NaN           NaN   \n",
       "1684                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "      genotype6_GQ  genotype6_PL  genotype7_sample_id  genotype7_str  \\\n",
       "14             NaN           NaN                  NaN            NaN   \n",
       "187            NaN           NaN                  NaN            NaN   \n",
       "721            NaN           NaN                  NaN            NaN   \n",
       "1176           NaN           NaN                  NaN            NaN   \n",
       "1684           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "      genotype7_num_alt  genotype7_allele_balance  genotype7_AD  genotype7_DP  \\\n",
       "14                  NaN                       NaN           NaN           NaN   \n",
       "187                 NaN                       NaN           NaN           NaN   \n",
       "721                 NaN                       NaN           NaN           NaN   \n",
       "1176                NaN                       NaN           NaN           NaN   \n",
       "1684                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "      genotype7_GQ  genotype7_PL  genotype8_sample_id  genotype8_str  \\\n",
       "14             NaN           NaN                  NaN            NaN   \n",
       "187            NaN           NaN                  NaN            NaN   \n",
       "721            NaN           NaN                  NaN            NaN   \n",
       "1176           NaN           NaN                  NaN            NaN   \n",
       "1684           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "      genotype8_num_alt  genotype8_allele_balance  genotype8_AD  genotype8_DP  \\\n",
       "14                  NaN                       NaN           NaN           NaN   \n",
       "187                 NaN                       NaN           NaN           NaN   \n",
       "721                 NaN                       NaN           NaN           NaN   \n",
       "1176                NaN                       NaN           NaN           NaN   \n",
       "1684                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "      genotype8_GQ  genotype8_PL  genotype9_sample_id  genotype9_str  \\\n",
       "14             NaN           NaN                  NaN            NaN   \n",
       "187            NaN           NaN                  NaN            NaN   \n",
       "721            NaN           NaN                  NaN            NaN   \n",
       "1176           NaN           NaN                  NaN            NaN   \n",
       "1684           NaN           NaN                  NaN            NaN   \n",
       "\n",
       "      genotype9_num_alt  genotype9_allele_balance  genotype9_AD  genotype9_DP  \\\n",
       "14                  NaN                       NaN           NaN           NaN   \n",
       "187                 NaN                       NaN           NaN           NaN   \n",
       "721                 NaN                       NaN           NaN           NaN   \n",
       "1176                NaN                       NaN           NaN           NaN   \n",
       "1684                NaN                       NaN           NaN           NaN   \n",
       "\n",
       "      genotype9_GQ  genotype9_PL  \n",
       "14             NaN           NaN  \n",
       "187            NaN           NaN  \n",
       "721            NaN           NaN  \n",
       "1176           NaN           NaN  \n",
       "1684           NaN           NaN  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merk[merk[\"gene\"].apply(lambda i: i in [ibd.node[ID][\"name\"] for ID in path])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  },
  "notify_time": "5",
  "toc": {
   "toc_cell": true,
   "toc_number_sections": true,
   "toc_threshold": 6,
   "toc_window_display": true
  },
  "toc_position": {
   "height": "413px",
   "left": "1893.38px",
   "right": "20px",
   "top": "193px",
   "width": "489px"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
